app/services/find_empty_trainees.rb

```diff
@@ -4,72 +4,96 @@
 class FindEmptyTrainees
   include ServicePattern
 
-  EXCLUDED_FIELDS = %w[
-    created_at
-    updated_at
-    dttp_update_sha
-    progress
-    slug
-    training_route
-    id
-    provider_id
-    ebacc
+  FIELDS_TO_CHECK = %w[
```

Comment:
Would this be easier to maintain if we kept the `EXCLUDED_FIELDS` list and used `Trainee.column_names - EXCLUDED_FIELDS` or something like that to build the query. Just thinking we're more less likely to add fields that we want to exclude.

Reply:
So i thought about doing it that way first but then we run into an issue when field names are ambiguous . For eg https://github.com/DFE-Digital/register-trainee-teachers/pull/1539/files#diff-d61925cb38bb78ca51a15ac7c23a6f89f581a5df3dd460a72295c10abf5d3690R18 and https://github.com/DFE-Digital/register-trainee-teachers/pull/1539/files#diff-d61925cb38bb78ca51a15ac7c23a6f89f581a5df3dd460a72295c10abf5d3690R50 are also fields in other tables as part of the query. Explicitly listing the fields here would give us a little control over deciding which fields to query.

We could probably get around it by renaming the field names in the other tables but just thinking about trade offs

Reply:
Could we add the table name when we build the SQL?  `trainees.<whatevs>`

Reply:
done https://github.com/DFE-Digital/register-trainee-teachers/pull/1539/commits/99f218e1b989728637e85420aa3eb523791a3b44